### ISSUE TITLE
Create monorepo and expose individual components

### ISSUE THREAD

**Issue by vjpr** [smeijer/unimported#37]
It would be really nice to have a good npm package to find all imports and requires. There are a few that exist, but they are not bulletproof.

- find-imports
- import-scan
- etc.

**Comment by smeijer** [comment]
Thanks for the idea! I have two questions if you don't mind:

1. Do you have a use case for this package? It helps me get an idea of how it would be used, and thereby I'd be better able to shape the API.

2. What is it that `find-imports` or `import-scan` don't do properly, that you expect this package to do better?

**Comment by vjpr** [comment]
> `import-scan`

Uses a simple regex. Captures `require('${__dirname}')` statements. Not as robust I guess.

> `find-imports`

Uses webpack which can fail on some complicated codebases.

> 1. Do you have a use case for this package? 

I wrote in another issue.

I want to see what deps are used in code but not declared in package.json. Then I generate code to fix these issues in https://pnpm.io/pnpmfile. This is also a common problem for `yarn pnp`.

**Comment by smeijer** [comment]
Yeah, it starts to make sense to me. It's not necessarily a separate package, but more that `unimported` can be used programmatically. So that you get the data returned, but without it printing to the console.

https://github.com/smeijer/unimported/blob/a9cd040351d82fbabfb6a6dc2213d2f18333d926/src/index.ts#L158-L176

That would be a nice feature to have.

**Comment by smeijer** [comment]
I'm going to close this, as I'm not able to dedicate time to it. Happy to accept pull-request that improve the public api though.
